rnd/autogpt_server/autogpt_server/server/server.py

```diff
@@ -18,48 +20,93 @@ def __init__(self, queue: execution.ExecutionQueue):
+            endpoint=AgentServer.schedule_agent,
             methods=["POST"],
         )
-        self.app.include_router(self.router)
-        self.app.on_event("startup")(db.connect)
-        self.app.on_event("shutdown")(db.disconnect)
+        router.add_api_route(
+            path="/agents/{agent_id}/schedules",
+            endpoint=AgentServer.get_execution_schedules,
+            methods=["GET"],
+        )
+
+        app.include_router(router)
+        app.on_event("startup")(db.connect)
+        app.on_event("shutdown")(db.disconnect)
+        uvicorn.run(app)
 
-    async def execute_agent(self, agent_id: str, node_input: dict):
-        agent = await graph.get_graph(agent_id)
+    @staticmethod
+    def get_agent(agent_id: str) -> graph.Graph:
+        agent = graph.get_graph(agent_id)
+        if not agent:
+            raise HTTPException(status_code=404, detail=f"Agent #{agent_id} not found.")
+
+        return agent
+
+    @staticmethod
+    def execute_agent(agent_id: str, node_input: dict) -> dict:
+        agent = graph.get_graph(agent_id)
         if not agent:
             raise HTTPException(status_code=404, detail=f"Agent #{agent_id} not found.")
 
         run_id = str(uuid.uuid4())
-        tasks = []
+        executions = []
 
         # Currently, there is no constraint on the number of root nodes in the graph.
         for node in agent.starting_nodes:
-            block = await node.block
+            block = node.block
             if error := block.input_schema.validate_data(node_input):
                 raise HTTPException(
                     status_code=400,
                     detail=f"Input data doesn't match {block.name} input: {error}",
                 )
 
-            task = execution.add_execution(
-                execution.Execution(
-                    run_id=run_id, node_id=node.id, data=node_input
-                ),
-                self.execution_queue,
+            executor_manager = get_service_client(ExecutionManager)
+            obj = executor_manager.add_execution(
+                run_id=run_id, node_id=node.id, data=node_input
             )
+            executions.append(obj)
 
-            tasks.append(task)
+        return {
+            "run_id": run_id,
+            "executions": executions,
+        }
+
+    @staticmethod
+    def get_executions(agent_id: str, run_id: str) -> list[execution.ExecutionResult]:
+        agent = graph.get_graph(agent_id)
+        if not agent:
+            raise HTTPException(status_code=404, detail=f"Agent #{agent_id} not found.")
 
-        return await asyncio.gather(*tasks)
+        return execution.get_executions(run_id)
 
+    @staticmethod
+    def schedule_agent(agent_id: str, cron: str, input_data: dict) -> dict:
+        executor_manager = get_service_client(ExecutionScheduler)
+        return {
+            "id": executor_manager.add_execution_schedule(agent_id, cron, input_data)
+        }
```

Comment:
This is the example of calling a "microservice", you just need to call `get_service_client(YOUR_SERVICE)` then call it like a normal function.